master/buildbot/www/avatar.py

```diff
@@ -51,6 +55,17 @@ def __init__(self,
         if github_api_endpoint is None:
             self.github_api_endpoint = self.DEFAULT_GITHUB_API_URL
         self.token = token
+        self.client_creds = None
+        if bool(client_id) != bool(client_secret):
+            config.error('client_id and client_secret must be both provided or none')
+        if client_id:
+            if token:
+                config.error('client_id and client_secret must not be provided when token is')
+            self.client_creds = base64.b64encode(b':'.join(
+                cred.encode('utf-8') for cred in (client_id, client_secret)
+            )).decode('ascii')
+        self.client_id = client_id
```

Comment:
I think we don't need to store the client_id and client_secret as members because we don't use them.

Sorry for asking for another iteration of the PR.

Reply:
Indeed, it was a leftover of a previous version of my change.